### PR TITLE
add whitelisted as array, add test for proxyPayment happy path

### PR DESCRIPTION
**Pull request by rstormsf**
1. Allow contract to buy tokens
2. Disable fallback for AIX contract
3. Add whitelist /blacklist as array
4. Add onlyController to whitelisting
5. Rename ETH to Wei

### CODE REVIEW

File: contracts/Contribution.sol [hunk 1]
```diff
@@ -113,13 +108,39 @@ contract Contribution is Controlled, TokenController {
...
+      address investorAddress = _investors[i];
+      if(canPurchase[investorAddress]) {
+        continue;
+      }
+      canPurchase[investorAddress] = true;
+      numWhitelistedInvestors++;
+    }
+  }
+
+  function whitelist(address investor) public onlyController {
```
Comment (Janther) [inline contracts/Contribution.sol]: this wasn't `onlyController` before 😱?

File: contracts/Contribution.sol [hunk 2]
```diff
@@ -22,8 +22,6 @@ contract Contribution is Controlled, TokenController {
   mapping (address => bool) public canPurchase;
   mapping (address => uint256) public individualEthCollected;
 
-  uint256 public minimum_investment;
```
Comment (Janther) [inline contracts/Contribution.sol]: This was not needed as you could set the minimum to `0` but it's ok

File: contracts/Contribution.sol [hunk 3]
```diff
@@ -113,13 +108,39 @@ contract Contribution is Controlled, TokenController {
     Initialized(initializedBlock);
   }
 
-  function whitelist(address investor) {
+  /// @notice interface for founders to blacklist investors
+  /// @param _investors array of investors
+  function blacklistAddresses(address[] _investors) public onlyController {
+    for (uint256 i = 0; i < _investors.length; i++) {
+      address investorAddress = _investors[i];
```
Comment (Janther) [inline contracts/Contribution.sol]: could we move this logic and call the `whitelist(address investor)` function?

Comment (Janther) [reply contracts/Contribution.sol]: perhaps a modifier called `onlyControllerOrSelf` could be implemented here

File: test/contribution_test.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const BigNumber = web3.BigNumber;
 import { expectThrow } from './utils.js';
 
 function getTime() {
-  return Math.floor(Date.now() / 1000);
+  return web3.eth.getBlock('latest').timestamp;
```
Comment (Janther) [inline test/contribution_test.js:11]: 👍

File: test/utils.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+const assert = require("chai").use(require('chai-as-promised')).assert;
```
Comment (Janther) [inline test/utils.js:1]: ❤️

**Review by Janther — APPROVED** [review]

File: contracts/Contribution.sol (hunk 1 above)
Comment (rstormsf) [reply contracts/Contribution.sol]: yeah

File: contracts/Contribution.sol (hunk 3 above)
Comment (rstormsf) [reply contracts/Contribution.sol]: `onlyControllerOrSelf` what do you mean? 
1. I agree, I can call `whitelist` method

**Review by Janther — APPROVED** [review]

**Pull request merged**